### PR TITLE
fix: [DPMMA-2661] Correct form mapper Mautic 5

### PR DESCRIPTION
**Pull request by tomekkowalczyk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 

## Description
Bug improves pool mapping via API Endpoint /api/forms/{form_id}/edit
The error causes the form fields to not be mapped correctly to the contact or company.
An incorrectly mapped field caused contacts to remain anonymous.

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send via API from /api/forms/{form_id}/edit with mapped field (leadField - deprecated or pair mappeField, mappedObject)

### CODE REVIEW

File: app/bundles/FormBundle/Tests/Controller/Api/FormApiControllerFunctionalTest.php [hunk 1]
```diff
@@ -126,29 +115,154 @@ public function testFormWorkflow(): void
...
+                            'alias'        => 'companyaddress1',
+                            'mappedField'  => 'companyemail',
+                            'mappedObject' => 'company',
+                            'leadField'    => 'companyemail',
+                        ],
+                        [
+                            'label'        => 'Company Phone',
+                            'type'         => 'tel',
+                            'alias'        => 'phone',
+                            'leadField'    => 'position',
```
Comment (patrykgruszka) [inline app/bundles/FormBundle/Tests/Controller/Api/FormApiControllerFunctionalTest.php]: Labels and aliases are confusing here, let's change it to match the mapped field

**Review by patrykgruszka — CHANGES_REQUESTED** [review]
Nice catch! The code looks solid. I've added a suggestion to make the tests clearer. Also, please address the CS Fixer issue - it seems like a minor fix.

File: app/bundles/FormBundle/Tests/Controller/Api/FormApiControllerFunctionalTest.php (hunk 1 above)
Comment (tomekkowalczyk) [reply app/bundles/FormBundle/Tests/Controller/Api/FormApiControllerFunctionalTest.php]: that's right, already corrected

**Review by patrykgruszka — APPROVED** [review]
Looks good now, thanks!

**Comment by andersonjeccel** [comment]
Should it be merged as bug fix? (5.1)
It seems like it’s targeting the wrong branch (5.x)

**Comment by patrykgruszka** [comment]
@andersonjeccel you're right. This should be rebased on 5.1

**Comment by tomekkowalczyk** [comment]
I change PR https://github.com/mautic/mautic/pull/13938.

**Pull request closed without merging**